2021-05-01 01:16:48,293 INFO     ---------------------------------------------------------------------------------------------
2021-05-01 01:16:48,294 INFO     Geo: cqd
2021-05-01 01:16:48,294 INFO     Data Path: data/NELL-betae
2021-05-01 01:16:48,294 INFO     #entity: 63361
2021-05-01 01:16:48,295 INFO     #relation: 400
2021-05-01 01:16:48,295 INFO     #max steps: 1000
2021-05-01 01:16:48,295 INFO     Evaluate unoins using: DNF
2021-05-01 01:16:48,295 INFO     loading data
2021-05-01 01:17:02,282 INFO     Training info:
2021-05-01 01:17:02,285 INFO     Validation info:
2021-05-01 01:17:02,286 INFO     3p: 4000
2021-05-01 01:17:02,287 INFO     Test info:
2021-05-01 01:17:02,287 INFO     3p: 4000
2021-05-01 01:17:03,628 INFO     Model Parameter Configuration:
2021-05-01 01:17:03,629 INFO     Parameter embeddings.0.weight: torch.Size([63361, 2000]), require_grad = True
2021-05-01 01:17:03,629 INFO     Parameter embeddings.1.weight: torch.Size([400, 2000]), require_grad = True
2021-05-01 01:17:03,629 INFO     Parameter Number: 127522000
2021-05-01 01:17:06,532 INFO     Loading checkpoint models/nell-betae...
2021-05-01 01:17:07,587 INFO     tasks = 3p
2021-05-01 01:17:07,587 INFO     init_step = 99999
2021-05-01 01:17:07,587 INFO     batch_size = 1000
2021-05-01 01:17:07,587 INFO     hidden_dim = 1000
2021-05-01 01:17:07,587 INFO     gamma = 12.000000
2021-05-01 01:17:07,587 INFO     Evaluating on Valid Dataset...
  0%|          | 0/4000 [00:00<?, ?it/s]2021-05-01 01:17:07,780 INFO     Evaluating the model... (0/4000)
  0%|          | 1/4000 [00:00<12:45,  5.23it/s]
main.py --do_valid --do_test --data_path data/NELL-betae -n 1 -b 1000 -d 1000 -lr 0.1 --disable_warmup --max_steps 1000 --cpu_num 0 --geo cqd --valid_steps 20 --tasks 3p --print_on_screen --test_batch_size 1 --optimizer Adagrad --reg_weight 0.05 --log_steps 5 --checkpoint_path models/nell-betae --cqd d2 --cqd-t-norm min --cqd-normalize --cqd-k 64 --no-save --cuda
overwritting args.save_path
logging to models/nell-betae
99999
Traceback (most recent call last):
  File "main.py", line 508, in <module>
    main(parse_args())
  File "main.py", line 497, in main
    valid_all_metrics = evaluate(model, valid_easy_answers, valid_hard_answers, args, valid_dataloader, query_name_dict, 'Valid', step, writer)
  File "main.py", line 165, in evaluate
    metrics = KGReasoning.test_step(model, tp_answers, fn_answers, args, dataloader, query_name_dict)
  File "/home/pminervi/workspace/KGReasoning/models.py", line 654, in test_step
    _, negative_logit, _, idxs = model(None, negative_sample, None, batch_queries_dict, batch_idxs_dict)
  File "/home/pminervi/anaconda3/envs/gpu/lib/python3.6/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/pminervi/workspace/KGReasoning/cqd.py", line 356, in forward
    k=self.k, t_norm=t_norm)
  File "/home/pminervi/workspace/KGReasoning/d2.py", line 158, in query_3p
    atom1_scores_3d = atom1_scores_3d.repeat(1, atom3_scores_3d.shape[1] // atom1_scores_3d.shape[1], 1)
RuntimeError: CUDA out of memory. Tried to allocate 992.00 MiB (GPU 0; 11.91 GiB total capacity; 9.87 GiB already allocated; 398.94 MiB free; 10.76 GiB reserved in total by PyTorch)
